### PR TITLE
Added Basic Reservoir Model

### PR DESCRIPTION
**Pull request by santacodes**
+ Added `basic_reservoir.py` adopted from Ferran's draft PR - pybamm-team/PyBaMM/pull/4184.
+ Added the `model entry` in `pyproject.toml`.
+ Changed the file naming conventions of models to abide by the ones in `PyBaMM`.

Subtask - #26

### CODE REVIEW

**Review by Saransh-cpp — APPROVED** [review]
Thanks, @santacodes! Looks good to me, but I'll wait for a review from @brosaplanella.

File: pyproject.toml
Comment (Saransh-cpp) [inline pyproject.toml:69]: Do we even need these in the project now?

Comment (santacodes) [reply pyproject.toml:69]: I think we could start removing things from the project when things inside the template get stable. At the end of the day, we will deliver the template to the users and not the project, so there is not much of a reason to maintain and test the project, other than for the template generation tests and maybe prototyping.

Comment (Saransh-cpp) [reply pyproject.toml:69]: Sounds good, thanks!

**Review by brosaplanella — APPROVED** [review]

File: pyproject.toml
Comment (agriyakhetarpal) [inline pyproject.toml:69]:
```suggestion
SPM = "pybamm_cookiecutter.models.input.spm:SPM"
BasicReservoir = "pybamm_cookiecutter.models.input.basic_reservoir:BasicReservoir"
```

File: {{cookiecutter.project_name}}/pyproject.toml
Comment (agriyakhetarpal) [inline {{cookiecutter.project_name}}/pyproject.toml:85]:
```suggestion
BasicReservoir = "{{ cookiecutter.__project_slug }}.models.input.basic_reservoir:BasicReservoir"
```

**Review by agriyakhetarpal — APPROVED** [review]
Looks great! I hope my suggestions below are in line with what you discussed you had to do in today's meeting.

File: {{cookiecutter.project_name}}/pyproject.toml
Comment (arjxn-py) [reply {{cookiecutter.project_name}}/pyproject.toml]: Just merged your suggestions hoping that it'd fix the failures

Comment (santacodes) [reply {{cookiecutter.project_name}}/pyproject.toml]: To fix the failures we would have to change the model names to the same ones as their keys in entry points. I will do it real quick!

Comment (santacodes) [reply {{cookiecutter.project_name}}/pyproject.toml]: It should work now, the assertion was failing because there was a list comparison and the lists weren't sorted so the unordered elements were causing the assertion to fail.

**Pull request merged**
